internal/provider/inventory_resource_test.go

```diff
@@ -186,20 +188,44 @@ func TestAccInventoryResource(t *testing.T) {
+					testAccCheckInventoryResourceExists("aap_inventory.test", &inventory),
 					testAccCheckInventoryResourceValues(&inventory, randomName, "", ""),
-					checkBasicInventoryAttributes(t, resourceNameInventory, randomName, "1", "Default"),
+					resource.TestCheckResourceAttr("aap_inventory.test", "name", randomName),
+					resource.TestCheckResourceAttr("aap_inventory.test", "organization", "1"),
+					resource.TestCheckResourceAttr("aap_inventory.test", "organization_name", "Default"),
+					resource.TestCheckResourceAttrSet("aap_inventory.test", "named_url"),
+					resource.TestCheckResourceAttrSet("aap_inventory.test", "id"),
+					resource.TestCheckResourceAttrSet("aap_inventory.test", "url"),
 				),
 			},
-			// Update and Read testing
+			// Update with new org and Read testing
+			{
+				Config: testAccInventoryResourceWithOrg(updatedName, updatedOrganizationId),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckInventoryResourceExists("aap_inventory.test", &inventory),
+					testAccCheckInventoryResourceValues(&inventory, updatedName, updatedDescription, updatedVariables),
+					resource.TestCheckResourceAttr("aap_inventory.test", "name", updatedName),
+					resource.TestCheckResourceAttr("aap_inventory.test", "organization", updatedOrganizationId),
+					resource.TestCheckResourceAttr("aap_inventory.test", "description", updatedDescription),
+					resource.TestCheckResourceAttr("aap_inventory.test", "variables", updatedVariables),
+					resource.TestCheckResourceAttr("aap_inventory.test", "named_url", fmt.Sprintf("/api/controller/v2/inventories/%s++%s/", updatedName, "test")),
+					resource.TestCheckResourceAttrSet("aap_inventory.test", "id"),
+					resource.TestCheckResourceAttrSet("aap_inventory.test", "url"),
+				),
+			},
+			// Update without new org and Read testing
 			{
 				Config: testAccInventoryResourceComplete(updatedName),
 				Check: resource.ComposeAggregateTestCheckFunc(
-					testAccCheckInventoryResourceExists(resourceNameInventory, &inventory),
+					testAccCheckInventoryResourceExists("aap_inventory.test", &inventory),
 					testAccCheckInventoryResourceValues(&inventory, updatedName, updatedDescription, updatedVariables),
-					checkBasicInventoryAttributes(t, resourceNameInventory, updatedName, "1", "Default"),
-					resource.TestCheckResourceAttr(resourceNameInventory, "description", updatedDescription),
-					resource.TestCheckResourceAttr(resourceNameInventory, "variables", updatedVariables),
+					resource.TestCheckResourceAttr("aap_inventory.test", "name", updatedName),
+					resource.TestCheckResourceAttr("aap_inventory.test", "organization", "1"),
+					resource.TestCheckResourceAttr("aap_inventory.test", "description", updatedDescription),
+					resource.TestCheckResourceAttr("aap_inventory.test", "variables", updatedVariables),
+					resource.TestCheckResourceAttr("aap_inventory.test", "named_url", fmt.Sprintf("/api/controller/v2/inventories/%s++%s/", updatedName, "Default")),
```

Comment:
Once we get the release completed, we will need to re-visit this part of the test.